### ISSUE TITLE
Empty the "Replace with" box when "Find what" changes

### ISSUE THREAD

**Issue by alexhenzell** [notepad-plus-plus/notepad-plus-plus#505]
When you change the value of "Find what" it's easy to forget to change "Replace with".

This is especially easy to do when replacing multiple variables with similar names.

For example, if you intend to do:

*Find what: positionX*
*Replace with: posX*

*Find what: positionY*
*Replace with: posY*

It would be easy to instead do:

*Find what: positionX*
*Replace with: posX*

*Find what: positionY*
*Replace with: posX*

One possible solution would be to automatically empty the "Replace with" box when "Find what" is altered, and provide an easy way of restoring the previous content of "Replace with" (such as a "Previous" button).

**Comment by DavidBailey100** [comment]
I agree, it is quite rare to want to search for something else, and not want to change the replacement - and it can be very dangerous as the OP illustrates. I would suggest the best thing would be offer a list of strings and replacement string <b>pairs<b> but wipe out the current replacement almost as the OP suggests.

However, leaving an empty replacement string isn't very safe either! Perhaps the replacement string should be $£$£$£$£$£$£$£ - i.e. something that found and corrected even after a careless replacement had been made and NP++ closed (because otherwise the answer is simply Undo!

**Comment by h-h-h-h** [comment]
@DavidBailey100:
> I agree, it is quite rare to want to search for something else, and not want to change the replacement

You might just want to alter the replacement string. I don't like the proposed feature as default functionality. :-1: Maybe, you both only have scenarios in mind where you always type words in the text boxes. There are many other scenarios with more complex expressions and replacement strings.
